### PR TITLE
Add support for optional iCheck on certain checkbox and radio inputs

### PR DESCRIPTION
**Pull request by kappalys**
I am targeting this branch, because I do not introduce breaking changes, and conserve backward compatibility.

## Changelog
```markdown
### Added
- Mechanic to disable iCheck on certain checkbox and radio inputs
```
## Subject
iCheck and Vue conflicts. To fix this we thought that it would be nice to introduce the same kind of mechanic that we can find on select2. We've added the 'data-sonata-icheck' attribute, which when set at false, will disable the iCheck mechanic on a specific input.

### CODE REVIEW

File: src/Resources/public/Admin.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ var Admin = {
         if (window.SONATA_CONFIG && window.SONATA_CONFIG.USE_ICHECK) {
             Admin.log('[core|setup_icheck] configure iCheck on', subject);
 
-            jQuery("input[type='checkbox']:not('label.btn>input'), input[type='radio']:not('label.btn>input')", subject).iCheck({
+            jQuery("input[type='checkbox']:not('label.btn>input, [data-sonata-icheck='false']'), input[type='radio']:not('label.btn>input')", subject).iCheck({
```
Comment (gimler) [inline src/Resources/public/Admin.js]: I think the code is not correct i have created a fiddle with tests

problems:
* invalid css selector
* not working for radios

https://jsfiddle.net/jffjgbem/9/

the correct one should:
`input[type='checkbox']:not(label.btn>input, [data-sonata-icheck='false']), input[type='radio']:not(label.btn>input, [data-sonata-icheck='false'])`

**Comment by jordisala1991** [comment]
I think this closes some issue when merged.

File: src/Resources/public/Admin.js (hunk 1 above)
Comment (kappalys) [reply src/Resources/public/Admin.js]: Hi, the single quotes syntax errors have indeed slipped through our proof reading process. This is due to the 3-level nesting of quotes, we'll patch it right now. The single quotes, in the `not` selectors, are not needed, we'll remove it at the same time.

However, we did not want to change the behaviour for the radio in the same pull request: we planned to add a new one if this one was merged.

**Comment by jordisala1991** [comment]
IMO adding a new feature should come with a doc update

**Comment by kappalys** [comment]
We checked the documentation, but couldn't find the page that mentions this particular set of features. Could you point to the right direction, please? Thank you.

**Comment by kunicmarko20** [comment]
@jordisala1991 isn't this a minor?

**Comment by kappalys** [comment]
Also, should we also include the feature for the radio?

**Comment by kunicmarko20** [comment]
As this is the same thing for checkbox/radio I would do it here and just adjust title and changelog.

**Comment by kappalys** [comment]
Alright, doing that ASAP

**Comment by kappalys** [comment]
Done

**Comment by kunicmarko20** [comment]
About the docs, there is mention here https://sonata-project.org/bundles/admin/3-x/doc/reference/action_create_edit.html#basic-configuration about iCheck or maybe a new entry in the cookbook as `select2` has its own also, wdyt @jordisala1991?

**Comment by jordisala1991** [comment]
One page like this one could be great: https://sonata-project.org/bundles/admin/3-x/doc/cookbook/recipe_select2.html

**Comment by kappalys** [comment]
Ok, will do that!

**Comment by OskarStark** [comment]
@kappalys could you finalize this PR, I am willing to merge this 

//cc @silasjoisten FYI

**Comment by kappalys** [comment]
@OskarStark We'll work on the documentation, but we would appreciate some help with this

**Comment by OskarStark** [comment]
> @OskarStark We'll work on the documentation, but we would appreciate some help with this

How can we help you?

**Comment by OskarStark** [comment]
Please create a file in the same directory like here:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/docs/cookbook/recipe_select2.rst

you can map the most of the select2 stuff to iCheck i guess, so this could be a good starting point 👍

**Comment by kappalys** [comment]
Error from me, 
the good option is : 
```php
'data-sonata-icheck' => 'false'
```

File: docs/cookbook/recipe_icheck.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/sonata_admin.yaml
+
+        sonata_admin:
+            options:
+                use_icheck:    false # disable iCheck
```
Comment (OskarStark) [inline docs/cookbook/recipe_icheck.rst]:
```suggestion
                use_icheck: false # disable iCheck
```

File: docs/cookbook/recipe_icheck.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Disable iCheck on some form elements
+-------------------------------------
+
+To disable iCheck on some ``checkbox`` or ``radio`` form elements, set data attribute ``data-sonata-icheck = "false"`` to this form element.
+
+.. code-block:: php
+
+    use Sonata\AdminBundle\Form\Type\ModelType;
```
Comment (OskarStark) [inline docs/cookbook/recipe_icheck.rst]:
```suggestion
    use Sonata\AdminBundle\Form\FormMapper;
    use Sonata\AdminBundle\Form\Type\ModelType;
```

**Comment by BitScout** [comment]
Is there anything preventing this from being merged?

**Comment by OskarStark** [comment]
Thank you

**Pull request merged**